### PR TITLE
1034 UI change capability details page

### LINKED ISSUE #1034: UI change Capability details page
Make UI changes for Capability details page.
See https://app.zeplin.io/project/62b9dba49dd2b211410e608e/screen/62b9dbf684718712f34fe8d0 for reference

### PR DESCRIPTION
**Pull request by minump**
## Description
UI changes as per Zeplin workspace is implemented for Capability-details view page.
Each style is given in classes named descriptively.

The capability view page is catalog/contributions/<contributionId>/capabilities/<capabilityId>

**Fixes #1034**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [x] Within a week
- [ ] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

**Comment by minump** [comment]
converting to draft to make 2 separate PRs.

**Comment by minump** [comment]
Have css style in h1, h2 tags instead of having separate class names

**Comment by minump** [comment]
changed css class names.
ready for review.

**Comment by minump** [comment]
ready for review @sandeep-ps @bingzhang

**Comment by sandeep-ps** [comment]
In the case of published capabilities, the View Contribution icon appears on the left side: 
image

Please check this case. This PR should address both published and unpublished capabilities, right?

**Comment by minump** [comment]
> In the case of published capabilities, the View Contribution icon appears on the left side: image
> 
> Please check this case. This PR should address both published and unpublished capabilities, right?

updated

File: contributions/catalog/webapps/templates/contribute/capability_details.html [hunk 1]
```diff
@@ -30,156 +26,175 @@
...
-        <div class="col-12 col-sm-2">
-            <h4>Is code open source?</h4>
-        </div>
-        <p class="col-sm-9">{{post.isOpenSource}}</p>
-    </div>
-
-    <div class="row">
-        <div class="col-12 col-sm-2">
-            <h4>Source code repository URL</h4>
+    <div class="grid container" style="display: grid; grid-auto-rows: max-content; row-gap: 5px; grid-template-rows: fit-content(100%)">
```
Comment (sandeep-ps) [inline contributions/catalog/webapps/templates/contribute/capability_details.html:29]: Hi, @minump, I got a better alignment with the following. Could you please take a look when you get a chance?

```
<div class="container">
        <div class="row">
            <div class="col-12 col-md-3">
                <h4>Capability Name </h4>
            </div>
            <div class="col-12 col-md-9">
                <div class="small-body-text">{{post.name}}</div>
            </div>
        </div>


        <div class="row">
            <div class="col-12 col-md-3">
                <h4>Capability description </h4>
            </div>
            <div class="col-12 col-md-9 ">
                <div class="small-body-text">{{post.description}}</div>
            </div>
        </div>
```

Similarly:

```
.small-body-text {
    font-size: 14px;
    height: 16px;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    letter-spacing: normal;
    margin: 15px auto;
    text-align: left;
    color: #404040;
}
```

Comment (sandeep-ps) [reply contributions/catalog/webapps/templates/contribute/capability_details.html:29]: Updated.

File: contributions/catalog/webapps/templates/contribute/capability_details.html [hunk 2]
```diff
@@ -30,156 +26,175 @@
...
+                            <td>{{post.dataDeletionEndpointDetails.description}}</td>
+                        </tr>
+                        {% endif %}
+                    </table>
+                </div>
+            </div>
+            {% endif %}
+            {% if post.deploymentDetails %}
+            <div class="row">
+                <div class="col-12 col-sm-3">
```
Comment (sandeep-ps) [inline contributions/catalog/webapps/templates/contribute/capability_details.html:140]:
```suggestion
                <div class="col-12 col-md-3">
```

There are other places where col-sm-* probably needs to be changed to col-md-*. Thanks.

File: contributions/catalog/webapps/templates/contribute/capability_details.html (hunk 1 above)
Comment (minump) [reply contributions/catalog/webapps/templates/contribute/capability_details.html]: With small body text ```margin : 11px auto;``` , the key and value does not seem to be in line (see screenshot). When its just ```margin :auto;``` it seems to be aligned.
Updated the rest of the styling as per feedback.
Screen Shot 2022-11-22 at 11 42 01 AM

File: contributions/catalog/webapps/templates/contribute/capability_details.html (hunk 2 above)
Comment (minump) [reply contributions/catalog/webapps/templates/contribute/capability_details.html]: updated

File: contributions/catalog/webapps/templates/contribute/capability_details.html (hunk 1 above)
Comment (sandeep-ps) [reply contributions/catalog/webapps/templates/contribute/capability_details.html]: With only auto margin, this is how it's getting rendered for me:

image

Have you tried refreshing without cache?

Comment (sandeep-ps) [reply contributions/catalog/webapps/templates/contribute/capability_details.html]: Hi, @minump, Could you please look into this when you get a chance?

Comment (minump) [reply contributions/catalog/webapps/templates/contribute/capability_details.html]: 11px seemed to be aligned towards the bottom for me. Maybe 7px? See attached screenshot for 7px. Updated css style margin for 7px.
Screen Shot 2022-11-29 at 4 51 03 PM

Comment (sandeep-ps) [reply contributions/catalog/webapps/templates/contribute/capability_details.html]: Okay. Looks close. Thanks.

**Review by sandeep-ps — APPROVED** [review]

**Review by sandeep-ps — APPROVED** [review]

**Pull request merged**